### DISCUSSION TITLE
Someone used some tricks to make his name appearing in my repo's `Used by` section

### DISCUSSION THREAD

**Discussion by louislam** [community/community#63362] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

@akirataguchi115 This guy added my repo in his `go.mod`. And now his name shows in my repo, but my repo is not a library or dependency, so there is only him in the list which is a bit wired.

https://github.com/akirataguchi115/akirataguchi115/blob/go.mod#L3605

```
github.com/louislam/uptime-kuma v0.0.0-20230404081326-be7d3f6142f4 // indirect
```

My suggestions:
1. Allow to hide `Used by` section
2. Check the dependencies are valid before adding to `Used by`? Since my repo is Javascript, adding as golang is impossible I think.

Someone already commented on his commit:
https://github.com/akirataguchi115/akirataguchi115/commit/e7cf62c66c06f0ce69f4a7482d9fb03677e8ac03#commitcomment-123925345

![](https://github.com/community/community/assets/1336778/68b20de8-63bd-465b-a6e8-cc71af3707e2)

**Comment by louislam** [comment]
cc: @SkwalExe

Since you commented on his commit.

**Comment by lambdakilo** [comment]
I think one way to fix this is disabling packages from your repository:
![image](https://github.com/community/community/assets/31825085/37cf0942-fec2-4281-91cc-10f6d431b021)

edit: We should patch this somehow. Don't know if disabling the importing of lang x package from lang y project would fix the issue. Importing JavaScript packages inside Go might not be possible but I'm pretty sure it's possible on other language mixups. 🦄

**Reply by louislam** [reply]
I don't think so. I have never enabled this.

![image](https://github.com/community/community/assets/1336778/b05a85ce-487e-4fda-987a-e6dc3be0a735)

**Comment by SkwalExe** [comment]
Disabling the "Used By" feature is impossible. 

![image](https://github.com/community/community/assets/69600456/daf794bb-c02a-43d6-9062-ca99e88a4e5f)

It can be abused, for example @akirataguchi115 by changing his username or repo's name could display arbitrary text on the uptime kuma's github page, which is a huge issue. 

I have added [a message about this on my readme](https://github.com/SkwalExe#if-you-saw-me-under-the-used-by-section-of-a-repo) one or two days ago, because I managed to put my profile on the official bash mirror repo. It is only a matter of time before starting to see people abusing this feature.

![image](https://github.com/community/community/assets/69600456/592b3748-975a-40c0-9220-54c01e23c848)

As a repo maintainer, you could try to put your own profile under the "used by" section of your repos so no one else can be directly shown.

**Reply by lambdakilo** [reply]
I agree this should be fixed. I'll take a look at the workarounds others have used when I get home from the pub.

**Reply by sheeit** [reply]
@akirataguchi115 you have a repo with a go.mod file that requires ~6000 repositories. Why did you do this?

**Reply by lambdakilo** [reply]
Good question!

**Comment by lambdakilo** [comment]
Somehow EbookFoundations has evaded the Used By hack. I remember putting myself in the homepage couple of months ago but they seem to have fixed this on their own now. Maybe ask how they solved it from one of the maintainers: https://github.com/EbookFoundation/free-programming-books ?

**Reply by SkwalExe** [reply]
Well you are still on their page. 

![image](https://github.com/community/community/assets/69600456/cfbf3212-abac-4f48-af0f-d94cb084f074)

There is just a bug where the "used by" section sometimes doesn't appear when you load the repo for the first time in your browser.

**Reply by lambdakilo** [reply]
Ohh that's true my bad completely forgot to spam refresh after the first load. Here should be a more valid example where I am not on the front page but should be: https://github.com/CyC2018/CS-Notes .

**Reply by SkwalExe** [reply]
This repo is not in your go.mod.

**Reply by lambdakilo** [reply]
Ahhhh my bad again, sorry! :( I thought I added 5000 most starred repos but I guess that one's fairly fresh. Good catch from your side. I guess we can conclude now that there's no apparent way of disabling this.

**Comment by Jenesius** [comment]
I Have the same problem with [https://github.com/Jenesius/vue-modal](https://github.com/Jenesius/vue-modal).
I found this package in the Ethernet: 
![image](https://github.com/community/community/assets/21982550/e629ad2d-a044-4046-9854-3047aab32699)
Maybe GitHub parse the npm(package store for Node) and adds all packages that link to my repository.

In result my **/network/dependets** looks like:
![image](https://github.com/community/community/assets/21982550/406b9291-1e80-40ce-a251-ee25ef7331a4)

I think this was not done intentionally by the user, but simply a packet parsing error.

**Reply by lambdakilo** [reply]
What was the error here? What was the intended behavior?

**Reply by Jenesius** [reply]
@akirataguchi115 I hope that when You will visit my package the field "UsedBy" will be disapled correct.
![image](https://github.com/community/community/assets/21982550/8fb7617c-ea13-4972-ae6b-e7fed0c5528b)
Now GitHub take info(count dep) from just first package.

This is a problem not related to you, I just stated such a problem. When support answers me, I will share with you

**Reply by lambdakilo** [reply]
Ohh now I see! This seems definitely like a problem. Could you open up another discussion in this organization? 🦄

**Comment by Jenesius** [comment]
Maybe for some one it can be usefull: [docs](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/exploring-the-dependencies-of-a-repository#changing-the-used-by-package)
